splash/render_options.py

```diff
@@ -27,54 +30,55 @@ def fromrequest(cls, request, max_timeout):
         """
 
         # 1. GET / POST data
-        data = {key: values[0] for key, values in request.args.items()}
-        if request.method == 'POST':
-            content_type = request.getHeader('content-type')
+        data = {key.decode('utf-8'): values[0].decode('utf-8') for key, values
+                in request.args.items()}
+        if request.method == b'POST':
+            content_type = request.getHeader(b'content-type')
             if content_type:
                 request.content.seek(0)
 
                 # 2. application/json POST data
-                if 'application/json' in content_type:
+                if b'application/json' in content_type:
                     try:
-                        data.update(json.load(request.content, encoding='utf8'))
+                        content = request.content.read().decode('utf-8')
+                        data.update(json.loads(content, encoding='utf8'))
                     except ValueError as e:
-                        raise BadOption("Invalid JSON: '{}'".format(e.message))
+                        raise BadOption("Invalid JSON: '{}'".format(str(e)))
 
                 # 3. js_source from application/javascript POST requests
-                if 'application/javascript' in content_type:
-                    data['js_source'] = request.content.read()
+                if b'application/javascript' in content_type:
+                    data['js_source'] = request.content.read().decode('utf-8')
                 request.content.seek(0)
 
         # 4. handle proxy requests
         if getattr(request, 'inspect_me', False):
             headers = [
-                (name, value)
+                (name.decode('utf-8'), value.decode('utf-8'))
                 for name, values in request.requestHeaders.getAllRawHeaders()
                 for value in values
-            ]
+                ]
             data.setdefault('headers', headers)
-            data.setdefault('http_method', request.method)
+            data.setdefault('http_method', request.method.decode('utf-8'))
 
             request.content.seek(0)
-            data.setdefault('body', request.content.read())
+            data.setdefault('body', request.content.read().decode('utf-8'))
```

Comment:
I'm not sure what implications does this change have for POST requests with binary bodies. I'm also not sure if we should care about it ;)